pom.xml

```diff
@@ -210,6 +210,21 @@ THE SOFTWARE.
                 <artifactId>maven-plugin</artifactId>
                 <version>3.7</version>
             </dependency>
+            <dependency>
+                <groupId>com.google.inject</groupId>
+                <artifactId>guice</artifactId>
+                <version>4.2.1</version>
+            </dependency>
+            <dependency>
+                <groupId>com.google.guava</groupId>
+                <artifactId>guava</artifactId>
+                <version>30.1-jre</version>
+            </dependency>
+            <dependency>
+                <groupId>org.slf4j</groupId>
+                <artifactId>slf4j-api</artifactId>
+                <version>1.7.30</version>
```

Comment:
should not be set here either - this is provided by jenkins and likely an older version.
if you use the version from the bom (set by the parent) at least your tests will blow up as oppsoed to tests passing then the plugin blowing up at runtime.

Whilst the plugin is using the `pluginFirstClassloader` the slf4j implementation coming from core will be a different (and older version) likely causing issues (if this is bundled, and if not - best not pretent that you have a version different to what you get)

Reply:
This sounds like the same logic would be applied to the `guava` dependency I had to add as well. The tests were failing because of the `guice` bump and I could only get it to pass by bumping guava (to at least `25.1-android`)

Reply:
so `guava` is probably a bit different as it is not split to a api implementation like `slf4j` is .